pkg/storaged/format-dialog.jsx

```diff
@@ -28,6 +28,148 @@ var StorageControls = require("./storage-controls.jsx");
 
 var _ = cockpit.gettext;
 
+function parse_options(options) {
+    if (options)
+        return (options.split(",")
+                       .map(function (s) { return s.trim() })
+                       .filter(function (s) { return s != "" }));
+    else
+        return [ ];
+}
+
+function has_option(options, opt) {
+    return parse_options(options).indexOf(opt) >= 0;
+}
+
+function adjust_options(options, opt, onoff) {
+    var split = parse_options(options);
+    var index = split.indexOf(opt);
+    if (onoff) {
+        if (index < 0)
+            split.push(opt);
+    } else {
+        if (index >= 0)
+            delete split[index];
+    }
+    return split.join(",");
+}
+
+function mounting_dialog_fields(is_custom, mount_dir, mount_options, visible) {
+
+    if (!visible)
+        visible = function () { return true; };
+
+    return [
+        { SelectOne: "mounting",
+          Title: _("Mounting"),
+          Options: [
+              { value: "default", Title: _("Default"), selected: !is_custom },
+              { value: "custom", Title: _("Custom"), selected: is_custom }
+          ],
+        },
+        { TextInput: "mount_point",
+          Title: _("Mount Point"),
+          Value: mount_dir,
+          visible: function (vals) {
+              return visible(vals) && vals.mounting == "custom";
+          },
+          validate: function (val) {
+              if (val.trim() == "")
+                  return _("Mount point can not be empty");
+          }
+        },
+        { TextInput: "mount_options",
+          Title: _("Mount Options"),
+          Value: mount_options == "defaults" ? "" : mount_options,
```

Comment:
So this parses an fstab option `defaults` to "" (which makes sense for the GUI); but shouldn't that be translated back once this gets written somewhere?

Reply:
This was already happening, check `format_dialog` and `maybe_update_config`.